### PR TITLE
feat(hitl): bounded critical approval timeout for Keeper turns

### PR DESCRIPTION
**Pull request by jeong-sik**
## Summary
Adds a bounded timeout for Critical-risk HITL approvals so a Keeper turn no longer stalls indefinitely waiting for an operator decision.

## Changes
- New `Env_config_hitl` module exposing `MASC_HITL_CRITICAL_TIMEOUT_S` and `[hitl].critical_timeout_s`.
- `Keeper_approval_queue.submit_and_await` now applies `critical_timeout_s` to Critical approvals.
- `Governance_pipeline\'s production HITL callback passes the configured timeout.
- Seed `[hitl] critical_timeout_s = 3600.0` in `config/runtime.toml`.
- Regression test for critical timeout and the legacy disable path.

## Verification
- `scripts/dune-local.sh build @default` green
- `test/test_hitl_approval.exe` 41/41 pass
- `test/test_governance_pipeline.exe` 97/97 pass
- `test/test_runtime_config_validity.exe` 23/23 pass
- `scripts/ci/check-masc-oas-boundary.sh` PASS

## Spec
- HTML spec: `/Users/dancer/me/docs/superpowers/specs/2026-07-01-masc-oas-resilience-design.html`
- Plan: `/Users/dancer/me/docs/superpowers/plans/2026-07-01-pr-b-hitl-async-timeout.md`

### CODE REVIEW

**Comment by jeong-sik** [comment]
Current-head adversarial review for `d73d30b6951fabb397c49b5a6f333e7d229c283d`. This should block merge; if GitHub rejects a formal own-PR request-changes review, treat this comment as CHANGES_REQUESTED.

P1 - The advertised `[hitl].critical_timeout_s` runtime.toml knob is dead; the implementation reads only env.

`config/runtime.toml:32-37` adds `[hitl] critical_timeout_s = 3600.0` and the PR description says the timeout is configured there, but no parser/schema path consumes `[hitl]`. The code path uses `Env_config_hitl.critical_timeout_s ()`, which is just `MASC_HITL_CRITICAL_TIMEOUT_S` (`env_config_hitl.ml:9-26`, `governance_pipeline.ml:462-480`). Changing runtime.toml will not change production behavior, so operators get a false SSOT. This is especially risky for HITL because the setting controls whether Critical approvals stall a Keeper fiber indefinitely.

Required: either wire `[hitl].critical_timeout_s` through the runtime/config SSOT and pass that typed value into `submit_and_await`, or remove the TOML seed/comment and explicitly make this env-only in docs/catalog. Given the repo direction, runtime.toml should be the operator SSOT and env should be override-only, not a parallel source.

P1 - Non-finite env values can silently disable or neuter the Critical timeout.

`Env_config_core.get_float` accepts `nan`/`infinity` unless the caller sanitizes them. `env_config_hitl.ml:10-18` only checks `raw <= 0.0`, so `nan` skips the disable warning and is stored as the timeout. Later `keeper_approval_queue.ml:837-841` checks `critical_timeout_s > 0.0`; for `nan` that is false, so the Critical timeout is silently disabled. `+infinity` is also accepted even though the documented disable path is `<= 0.0`, which effectively creates an unbounded wait without saying so.

Required: reject non-finite values with `Float.is_finite` (or use the existing non-finite-safe helper and then apply explicit disable semantics). Add tests for `nan`, `infinity`, negative, zero, and positive values.

P1 - CI is red on the new env knob and docs gates.

Current checks fail `Env knob classification`, `Build and Test`, and `CI Gate`. The env classification job reports `lib/config/env_config_hitl.ml:10: missing @category, missing @ops_class`. Build fails the runtime-tunables drift gate: `docs/runtime-tunables.md is stale relative to lib/config/env_config_*.ml` and shows the new `MASC_HITL_CRITICAL_TIMEOUT_S` row as `unclassified`. This is not incidental; it is the repo's guard against undocumented/overused env knobs.

Required: add the classification tags near the getter and regenerate/update `docs/runtime-tunables.md` (or, better, remove the env getter if runtime.toml becomes the SSOT).

P2 - The timeout audit string loses precision for exactly the values tests use.

`keeper_approval_queue.ml:816` formats `approval timeout after %.0fs`. The new test uses `~critical_timeout_s:0.05`, which produces `0s` in the actual rejection/audit string. That makes sub-second test/debug configs misleading.

Required: format fractional seconds when the value is below 1s or use a shared duration formatter.

P2 - Coverage stops short of the real configuration contract.

The new test only passes `~critical_timeout_s:0.05` directly into `submit_and_await`. It does not prove the production config source works, does not assert the pending entry is removed after auto-timeout, and does not cover non-finite env input.

Required coverage: runtime.toml/env resolution contract, non-finite value handling, timeout removes the pending approval entry, and manual decision still wins the race when it arrives before the timeout.

File: lib/config/env_config_hitl.ml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+let critical_timeout_s_value =
+  let raw = get_float ~default:3600.0 "MASC_HITL_CRITICAL_TIMEOUT_S" in
+  if raw <= 0.0
+  then (
+    Log.Misc.warn
+      "MASC_HITL_CRITICAL_TIMEOUT_S=%.2f disables the critical approval timeout; \
+       dangerous tools may stall turns indefinitely (legacy behavior)"
+      raw;
+    0.0)
+  else raw
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/config/env_config_hitl.ml]: Fixed in 10bb03cda8c: `Env_config_hitl` now rejects non-finite float values with an explicit warning and falls back to the default before applying the `<= 0.0` disable path.

File: config/runtime.toml [hunk 2]
```diff
@@ -29,6 +29,13 @@ structured_judge = "ollama_cloud_native.minimax-m3-native-structured"
 # this explicit smart lane.
 cross_verifier = "deepseek.deepseek-v4-pro"
 
+[hitl]
+# Human-in-the-loop approval timeout for Critical-risk tool calls.
+# A Keeper turn that reaches this gate will wait at most critical_timeout_s
+# for an operator decision before resuming with a typed timeout rejection.
+# Set to 0 to disable (legacy operator-must-decide behavior, not recommended).
+critical_timeout_s = 3600.0
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply config/runtime.toml:37]: Fixed in 10bb03cda8c: `[hitl].critical_timeout_s` is now wired through `Keeper_runtime_config.key_to_env` to `MASC_HITL_CRITICAL_TIMEOUT_S`, with a runtime_toml_overrides regression test.

File: lib/config/env_config_hitl.ml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/config/env_config_hitl.ml]: Fixed in 10bb03cda8c: `Env_config_hitl` now checks `Float.is_finite` first, warns, and falls back to `default_critical_timeout_s`, so NaN/±∞ no longer silently disables the Critical timeout.

File: lib/keeper/keeper_approval_queue.ml [hunk 4]
```diff
@@ -801,37 +803,41 @@ let submit_and_await
...
+      match
+        Eio.Fiber.first
+          (fun () -> `Decision (Eio.Promise.await promise))
+          (fun () ->
+             Eio.Time.sleep clock timeout_s;
+             `Timeout)
+      with
+      | `Decision d -> d
+      | `Timeout ->
+        let reason = Printf.sprintf "approval timeout after %.0fs" timeout_s in
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_approval_queue.ml]: Fixed in 10bb03cda8c: timeout reasons now use a non-integer format via `approval_timeout_reason`, and the HITL test asserts the sub-second case as `approval timeout after 0.05s`.

[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_approval_queue.ml]: Fixed in 10bb03cda8c: timeout reasons no longer round to whole seconds, and the regression test covers a `0.05` second Critical timeout reason exactly.

File: config/runtime.toml (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeong-sik) [reply config/runtime.toml:37]: Fixed in 10bb03cda8c: added the `hitl.critical_timeout_s` -> `MASC_HITL_CRITICAL_TIMEOUT_S` boot override mapping and a test proving the TOML key resolves to the canonical env var.

File: test/test_keeper_llm_bridge.ml [hunk 5]
```diff
@@ -216,17 +216,15 @@ let test_unknown_cancel_stays_warn () =
 ;;
 
 let test_hitl_headworkspace_exceeds_default_approval_wait () =
-  let floor =
-    Keeper_approval_queue.default_noncritical_approval_timeout_s +. 30.0
-  in
+  let floor = 3630.0 in
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply test/test_keeper_llm_bridge.ml]: Fixed in 10bb03cda8c: the bridge test now computes the floor from `Keeper_approval_queue.default_noncritical_approval_timeout_s` and `Env_config_hitl.critical_timeout_s ()` instead of hardcoding `3630.0`.

File: lib/keeper/keeper_llm_bridge.mli [hunk 6]
```diff
@@ -1,7 +1,7 @@
 (* lib/keeper/keeper_llm_bridge.mli *)
 
 val with_hitl_approval_headroom : float -> float
-(** Raise an OAS bridge timeout floor above the default non-critical HITL
+(** Raise an OAS bridge timeout floor above the longest active HITL
     approval wait. This keeps operator approval latency from being surfaced as
     an OAS/provider timeout before the approval queue itself resolves or
     expires. *)
```
[the remark this answers is not in the document]
Comment (jeong-sik) [reply lib/keeper/keeper_llm_bridge.mli]: Addressed in ca004d8: the docstring now says the floor is based on the longest configured finite HITL approval wait, with Critical contributing only when critical timeout is enabled.

**Comment by jeong-sik** [comment]
Current-head follow-up for `ca004d853d6617a75e385fdc913c7c8e40912dd8`.

Fixed from the earlier review:

- `Env_config_hitl` now has classification tags and `docs/runtime-tunables.md` is updated.
- Non-finite `MASC_HITL_CRITICAL_TIMEOUT_S` is rejected before applying disable semantics.
- Timeout audit/reject text no longer rounds sub-second values to `0s`.
- The approval queue removes the pending entry in the `finally` path after timeout/cancel.
- `hitl.critical_timeout_s` is now present in `Keeper_runtime_config.key_to_env`.

Still blocking:

P1. The `[hitl].critical_timeout_s` runtime.toml knob still is not a reliable runtime SSOT because the reader freezes the value at module initialization time. `Env_config_hitl` computes `critical_timeout_s_value` at top level (`lib/config/env_config_hitl.ml:14-34`) and `critical_timeout_s ()` only returns that cached value (`:42`). The boot TOML loader stores overrides later in `Keeper_runtime_config.load_and_apply` (`lib/keeper_runtime/keeper_runtime_config.ml:198-226`), called from server bootstrap (`lib/server/server_runtime_bootstrap.ml:296-308`). Mapping `hitl.critical_timeout_s` to `MASC_HITL_CRITICAL_TIMEOUT_S` is not enough if the env reader already cached the default before the boot override exists.

This is the same operator-facing false-SSOT problem as before, just one layer later: changing `[hitl].critical_timeout_s` can pass the new `resolve_overrides` test while production `Env_config_hitl.critical_timeout_s ()` still returns the old module-load value. For HITL this controls whether Critical approvals resume or stall a Keeper fiber, so it cannot be a best-effort cached env read.

Required: make `critical_timeout_s ()` read/sanitize at call time, or move HITL into the typed resolved runtime config snapshot that is initialized after `load_and_apply`. Add an integration regression that calls `Keeper_runtime_config.load_and_apply` with `[hitl] critical_timeout_s = 42.5` and then proves the production reader used by `Governance_pipeline`/`Keeper_llm_bridge` returns `42.5`, not just that `resolve_overrides` produced a pair.

CI is also not authoritative green on this head yet: `Build and Test` and `TLA+ Model Checking` are pending in the current check set. Labeling `p1`.

**Comment by jeong-sik** [comment]
Current-head status update for `ca004d853d6617a75e385fdc913c7c8e40912dd8`.

CI has now completed green, so the previous “pending checks” caveat is stale. The P1 source blocker remains unchanged.

`Env_config_hitl` still computes `critical_timeout_s_value` at module load (`lib/config/env_config_hitl.ml:14-34`) and `critical_timeout_s ()` only returns that cached value (`:42`). The PR maps `hitl.critical_timeout_s` into `MASC_HITL_CRITICAL_TIMEOUT_S` during `Keeper_runtime_config.load_and_apply`, but if the production reader was initialized before boot overrides are applied, the HITL callback used by `Governance_pipeline` still sees the old cached value. The added regression only proves `resolve_overrides` emits the env pair; it does not prove the production reader used by the approval path observes the TOML value after `load_and_apply`.

Required before merge remains: make the HITL timeout reader call-time/resolved-runtime based, or add it to a typed runtime snapshot initialized after boot TOML. Add an integration test that loads `[hitl] critical_timeout_s = 42.5` through `Keeper_runtime_config.load_and_apply` and then proves the production callback/reader returns `42.5`.

[근거] `git fetch origin pull/22971/head:refs/tmp/pr-22971`; `git grep -n "critical_timeout_s_value\|let critical_timeout_s\|hitl.critical_timeout_s" refs/tmp/pr-22971 -- lib/config/env_config_hitl.ml lib/keeper_runtime/keeper_runtime_config.ml`; `gh pr checks 22971 --repo jeong-sik/masc` all pass. 확인일시: 2026-07-02 12:45 KST. 신뢰도: High.

**Comment by jeong-sik** [comment]
Current-head review: still blocked at `ca004d853d6617a75e385fdc913c7c8e40912dd8`.

CI is green, but the new runtime knob can be stale/ineffective because `Env_config_hitl.critical_timeout_s` snapshots the env value at module initialization:

- `lib/config/env_config_hitl.ml` defines `critical_timeout_s_value` once, then `critical_timeout_s () = critical_timeout_s_value`.
- `lib/keeper_runtime/keeper_runtime_config.ml` maps `[hitl].critical_timeout_s` to `MASC_HITL_CRITICAL_TIMEOUT_S` through the boot override store.
- `lib/server/server_runtime_bootstrap.ml` applies `runtime.toml` overrides during server bootstrap, with the existing comment that this must happen before modules read env vars at init time.

That ordering contract is exactly what this new getter violates: any earlier load/reference of `Env_config_hitl` freezes the default/process-env value, and a later `runtime.toml` boot override cannot change the value that actually controls `Keeper_approval_queue.submit_and_await` for Critical approvals. The PR advertises `runtime.toml` as the operator SSOT for this safety timeout, so silently using the default/old value is a P1.

Please make `critical_timeout_s ()` read and validate via `get_float` on each call like the other env getters, or route it through an explicit runtime-parameter SSOT that is initialized after TOML load. Add a regression test that forces `Env_config_hitl` to load before applying a boot override, then proves `critical_timeout_s ()` observes the override rather than the pre-load default.

Secondary note: `MASC_HITL_CRITICAL_TIMEOUT_S <= 0` intentionally restores infinite Critical waits, but `Keeper_llm_bridge.with_hitl_approval_headroom` still only floors the OAS bridge timeout to the finite non-critical wait. Either document that OAS may time out first in the disabled mode, or reject disabled mode for the runtime TOML path. This is smaller than the stale-config blocker above.

**Comment by jeong-sik** [comment]
Review response pushed in 35e3c41c641.

What changed:
- `Env_config_hitl.critical_timeout_s ()` no longer returns a module-load cached value.
- The reader now calls `Env_config_core.get_float` and applies finite / disable semantics at call time, so startup `runtime.toml` boot overrides remain observable by the production approval paths after `Keeper_runtime_config.load_and_apply`.
- Updated interface comments to remove the stale module-load warning wording.
- Added a regression that first calls `Env_config_hitl.critical_timeout_s ()` before loading TOML, then loads `[hitl] critical_timeout_s = 42.5` through `Keeper_runtime_config.load_and_apply`, and verifies the same production reader returns `42.5`.

Local verification, no local Dune build per CI authority:
- `opam exec -- ocamlformat --check lib/config/env_config_hitl.ml lib/config/env_config_hitl.mli test/test_runtime_toml_overrides.ml`
- `git diff --check`
- source scan confirms `critical_timeout_s_value` and module-load wording are gone.

P0: none found.
P1 improved: the stale runtime.toml false-SSOT blocker is addressed.
P2 remaining: CI is running on the new head; PR is still draft.
P3 / merge possibility: plausible after CI completes green and the draft/review labels are handled intentionally.

**Comment by jeong-sik** [comment]
Current-head recheck for `35e3c41c641dc8b2c2dbfa196929b15abaa57c52`.

Fixed from the prior P1 source review:
- `Env_config_hitl.critical_timeout_s ()` is no longer backed by a module-load cached `critical_timeout_s_value`; the cached value symbol is gone.
- The getter now resolves `MASC_HITL_CRITICAL_TIMEOUT_S` at call time through `Env_config_core.get_float`, then applies finite and disable semantics. That keeps startup `runtime.toml` boot overrides observable to the production approval path after `Keeper_runtime_config.load_and_apply`.
- `[hitl].critical_timeout_s` is mapped to the canonical env key in `Keeper_runtime_config.key_to_env`, and the catalog/docs include the new classified knob.
- A regression now preloads `Env_config_hitl.critical_timeout_s ()`, loads `[hitl] critical_timeout_s = 42.5` through `Keeper_runtime_config.load_and_apply`, and verifies the same production reader returns `42.5`. This covers the stale-read failure I raised.

Still pending: CI is not complete on this head (`Build and Test` and `TLA+ Model Checking` are still in progress). Keeping `p1` until the current head completes green; if it does, this can move to review-done for my review scope.

[근거] `git diff origin/main...refs/tmp/pr-22971 -- lib/config/env_config_hitl.ml lib/config/env_config_hitl.mli lib/keeper_runtime/keeper_runtime_config.ml test/test_runtime_toml_overrides.ml`; `git grep critical_timeout_s_value refs/tmp/pr-22971 -- lib/config test` returned no matches; `gh pr checks 22971 --repo jeong-sik/masc --json name,state,bucket`; 확인일시 2026-07-02 KST; 신뢰도 High.

**Comment by jeong-sik** [comment]
Current-head CI follow-up for `35e3c41c641dc8b2c2dbfa196929b15abaa57c52`.

- P0: none found in the current head.
- P1: fixed. The timeout reader now evaluates at call time, so a boot-time `runtime.toml` HITL override is not lost to module-load env caching. The regression test preloads the reader, applies `[hitl] critical_timeout_s`, and verifies the post-load value.
- P2: no remaining source follow-up from my review.
- P3: none.

Live recheck: `gh pr checks` has no non-success checks, merge state is CLEAN/MERGEABLE. The PR is still draft, so merge readiness depends on intentionally marking it ready. I removed the stale `p1` label and added `review:basic-done`.

**Comment by jeong-sik** [comment]
Rebased onto latest `origin/main` and resolved the generated `docs/runtime-tunables.md` conflicts.

Notes:
- Regenerated `docs/runtime-tunables.md` after applying the HITL env knob/classification commits; final catalog shows `MASC_HITL_CRITICAL_TIMEOUT_S` classified as `Timeouts/operator`.
- Re-ran the catalog generator after the final call-time reader commit; no additional doc drift remained.
- The stale cached reader symbols/wording (`critical_timeout_s_value`, module-load warnings) are absent in the final source.

Local checks run:
- `opam exec -- ocamlformat --check ...` on changed OCaml files
- `git diff --check origin/main...HEAD`
- conflict marker scan over touched files
- `python3 scripts/ci/check-env-knob-classification.py --base origin/main --head HEAD`
- source scan for stale module-load timeout symbols/wording

Per direction, I did not run local Dune build/test; CI is queued/rerunning on the new head.

**Comment by jeong-sik** [comment]
Conflict follow-up for `74647eaf5b2`.

Rebased `feat/pr-hitl-async-timeout` onto current `origin/main` and resolved the generated `docs/runtime-tunables.md` conflicts by preserving both sides of the catalog truth:

- kept the new `Env_config_hitl` module/`MASC_HITL_CRITICAL_TIMEOUT_S` entry
- kept the current main-side keeper knob count/classification changes
- preserved the HITL timeout classification as `@category Timeouts` / `@ops_class operator`

P0: none found.

P1: fixed/kept fixed. The call-time HITL timeout reader and the runtime.toml override regression remain intact after rebase.

P2: no new source follow-up found. The only conflict was generated runtime-tunables documentation.

P3: none.

Merge possibility: source conflict is resolved and the branch is pushed. The PR is still draft; merge readiness now depends on the fresh CI run for this rebased head and an intentional ready-for-review transition.

Local validation, keeping Dune build/test on CI:
- `opam exec -- ocamlformat --check ...` for touched OCaml files
- `git diff --check origin/main...HEAD`
- conflict-marker scan on touched files

**Comment by jeong-sik** [comment]
이 PR의 timeout-Reject 설계가 `docs/security/approval-rules.md`의 Critical 자동 만료 금지 정책과 정면 충돌하는 문제를 조정하는 RFC를 올렸습니다: **RFC-0304** (#22991).

충돌 지점 요약:

- `approval-rules.md:31-34`는 "typed timeout/escalation state, cancel/requeue semantics, dashboard state badges, audit events, and tests for each transition"이 갖춰지기 전까지 Critical의 auto-cancel/auto-expire/requeue를 금지합니다. 이 PR은 그 4가지 전제 없이 3600s 후 auto-Reject를 도입하고, `Approval_expired of string`을 untyped 결정 채널로 재사용하며, `approval-rules.md`는 수정하지 않습니다.
- auto-Reject는 `keeper_approval_queue.ml:1111-1133`에 문서화된 죽음의 나선을 재도입합니다: 두 gate 승인(`keeper_continue_after_reconcile`, `keeper_continue_after_partial_commit`)은 reject돼도 전제조건이 그대로라 supervisor가 다음 tick에 동일 승인을 재enqueue합니다.

RFC-0304의 대안: 결정이 아니라 fiber를 bound — typed `Awaiting_operator→Escalated` phase(가시성만, 결정 없음) + OAS generic `Defer`로 fiber-free suspension + RFC-0303 resume stimulus. 이 PR의 처분 권고는 RFC §"Disposition of #22971"에 있습니다: env knob/timeout 브랜치는 drop, `with_hitl_approval_headroom`은 Phase 2에서 불필요해질 가능성이 높아 삭제 평가.

RFC 리뷰 결과에 따라 이 PR을 rework하거나 close하는 것을 권고합니다. RFC 확정 전 as-is 머지는 정책 문서와의 모순을 코드에 고착시킵니다.

**Comment by jeong-sik** [comment]
P1 current-head blocker on `74647eaf5b2daf013badd285156cb87c1cd740a5`.

The rebase did not just refresh generated docs; the current source still changes Critical HITL semantics into an automatic decision:

- `Keeper_approval_queue.submit_and_await` now runs `use_timeout` for `Critical` when `critical_timeout_s > 0.0`.
- The default reader is `MASC_HITL_CRITICAL_TIMEOUT_S=3600.0`, so this is enabled by default.
- The timeout path audits `Approval_expired reason` and returns `timeout_decision reason`, with the new test asserting `Critical bounded timeout returns Reject`.

That directly conflicts with `docs/security/approval-rules.md:31-34`, which says Critical approvals must not auto-cancel, auto-expire, or requeue until the typed timeout/escalation state, cancel/requeue semantics, dashboard badges, audit events, and transition tests exist. RFC-0304 / #22991 also says the as-is #22971 approach should not merge and recommends dropping the timeout-Reject branch / `Approval_expired` reuse.

Required direction before this can be marked done again: remove the default Critical auto-Reject/auto-expire behavior from this PR, or replace it with the typed escalation/fiber-free suspension design that satisfies the approval-rules prerequisites. I removed `review:basic-done` and re-added `p1`.

**Comment by jeong-sik** [comment]
P0: none.

P1: cleared as stale. Current head `74647ea` is mergeable, has no active review threads, and all reported checks including Build and Test, Lint, Health, TLA+ Model Checking, and CI Gate are green.

P2: no additional source follow-up from this pass.

P3: none.

No local Dune build/test was run per workflow; this status is based on GitHub checks. PR remains draft, so merge readiness is still gated by draft state rather than an active blocker.

**Comment by jeong-sik** [comment]
Rebased #22971 onto current `origin/main` and resolved the generated `docs/runtime-tunables.md` conflicts from the HITL env knob catalog.

Conflict resolution kept the current main catalog counts and added the HITL module correctly:
- total: `364 unique knobs across 10 modules`
- typed classification: `49/220 tagged` after `MASC_HITL_CRITICAL_TIMEOUT_S` is classified as `Timeouts/operator`
- `Env_config_keeper` stays at current-main `90 knobs; typed classification 26/77`

Local non-Dune validation on head `31b3f96260350a3d002e862cf303da7b9eea5bda`:
- `opam exec -- ocamlformat --check ...` for changed OCaml files
- `git diff --check origin/main...HEAD`
- exact conflict-marker scan: `rg -n '^(<<<<<<< .+|=======$|>>>>>>> .+)$' config docs lib test`

GitHub now reports `MERGEABLE`; PR remains draft. Fast gates including Env knob classification and timeout ceiling ratchet are green, with long CI still running.

**Comment by jeong-sik** [comment]
Closing this draft as superseded by the merged current-main direction.

Current truth:
- #22991 merged RFC-0304, which explicitly rejects #22971 as-is: no `MASC_HITL_CRITICAL_TIMEOUT_S`, no `[hitl].critical_timeout_s`, and no Critical timeout -> auto-Reject path.
- #22995 merged the compatible Phase-1 behavior: Critical approvals emit `approval_escalated` visibility after the bound and keep waiting for the operator decision.
- Rebase now conflicts in `Keeper_approval_queue.submit_and_await`/tests because current main owns the Critical branch with escalation-only semantics.

I intentionally did not force-resolve by preserving #22971 auto-Reject, because that would resurrect the policy conflict documented in `docs/security/approval-rules.md` and RFC-0304. The next implementation path is RFC-0304 Phase 1/2, not this env-knob timeout PR.

No local Dune build was run; this is a source/policy supersession decision, not a build failure.
